webpages/popup/index.js

```diff
@@ -132,3 +138,52 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
 });
 
 chrome.runtime.sendMessage("checkPermissions");
+
+// April Fools 2024 below
+
+class Haddock {
+  constructor(elem) {
+    this.elem = elem;
+    this.facingRight = parseInt(elem.style.left) > 200;
+    (async () => {
+      while (true) {
+        this.facingRight = !this.facingRight;
+        const duration = 6 + Math.random() * 5; // 6-11 secs
+        this.cycle(duration);
+        await new Promise((r) => setTimeout(r, duration * 1000));
+      }
+    })();
+  }
+
+  cycle(duration) {
+    this.elem.style.transitionProperty = "transform, left, top";
+    this.elem.style.transitionDuration = `0.2s, ${duration}s, ${duration}s`;
+    setTimeout(() => {
+      this.elem.style.transform = `scaleX(${this.facingRight ? -1 : 1})`;
+      // If the haddock is facing left, move to anywhere on the left half of the screen.
+      // If the haddock is facing right, move to anywhere on the right half of the screen.
+      this.elem.style.left = (Math.random() + this.facingRight) * (window.innerWidth / 2) + "px";
+      this.elem.style.top = Math.random() * window.innerHeight + "px";
+    }, 0);
+  }
+}
+
+function spawnFish(initX, initY) {
+  const fishElem = document.createElement("span");
+  fishElem.textContent = "🐟";
+  fishElem.style.pointerEvents = "none";
+  fishElem.style.position = "absolute";
+  fishElem.style.left = initX + "px";
+  fishElem.style.top = initY + "px";
+  fishElem.style.fontSize = "22px";
+  fishElem.style.textShadow = "#000 0px 2px 5px";
+  new Haddock(document.body.appendChild(fishElem));
+}
+
+const now = Date.now();
+if (true /* TODO */) {
```

Comment:
```suggestion
if (!window.matchMedia("(prefers-reduced-motion)").matches /* TODO */) {
```

Reply:
Thanks, I almost forgot

Reply:
If preferring reduced motion makes the swimming fish emojis disappear, then we should probably keep them in the header for those users.